packages/in_app_purchase/in_app_purchase_storekit/darwin/Classes/InAppPurchasePlugin+TestOnly.h

```diff
@@ -15,14 +15,18 @@
 // Callback channel to dart used for when a function from the transaction observer is triggered.
 @property(strong, nonatomic) FlutterMethodChannel *transactionObserverCallbackChannel;
 
+// Callback channel to dart used for when a function from the transaction observer is triggered.
+@property(strong, nonatomic) FIAPRequestHandler * (^handlerFactory)(SKRequest *);
```

Comment:
```suggestion
@property(copy, nonatomic) FIAPRequestHandler * (^handlerFactory)(SKRequest *);
```

You want to use `copy` for blocks properties since that will force any resources on the stack to be copied to the heap ( https://stackoverflow.com/questions/4081831/how-to-store-blocks-in-properties-in-objective-c#comment11631504_4081903 )